apps/ensnode/src/handlers/Resolver.ts

```diff
@@ -2,341 +2,349 @@ import { type Context } from "ponder:registry";
-  context: Context;
-  event: EventWithArgs<{ node: Node; interfaceID: Hex; implementer: Hex }>;
-}) {
-  const { node, interfaceID, implementer } = event.args;
-  const id = makeResolverId(event.log.address, node);
-  await upsertResolver(context, {
-    id,
-    domainId: node,
-    address: event.log.address,
-  });
-
-  // log ResolverEvent
-  await context.db.insert(schema.interfaceChanged).values({
-    ...sharedEventValues(event),
-    resolverId: id,
-    interfaceID,
-    implementer,
-  });
-}
-
-export async function handleAuthorisationChanged({
-  context,
-  event,
-}: {
-  context: Context;
-  event: EventWithArgs<{
-    node: Node;
-    owner: Hex;
-    target: Hex;
-    isAuthorised: boolean;
-  }>;
-}) {
-  const { node, owner, target, isAuthorised } = event.args;
-  const id = makeResolverId(event.log.address, node);
-  await upsertResolver(context, {
-    id,
-    domainId: node,
-    address: event.log.address,
-  });
-
-  // log ResolverEvent
-  await context.db.insert(schema.authorisationChanged).values({
-    ...sharedEventValues(event),
-    resolverId: id,
-    owner,
-    target,
-    // NOTE: the spelling difference is kept for subgraph backwards-compatibility
-    isAuthorized: isAuthorised,
-  });
-}
-
-export async function handleVersionChanged({
-  context,
-  event,
-}: {
-  context: Context;
-  event: EventWithArgs<{ node: Node; newVersion: bigint }>;
-}) {
-  // a version change nulls out the resolver
-  const { node, newVersion } = event.args;
-  const id = makeResolverId(event.log.address, node);
-  const domain = await context.db.find(schema.domain, { id: node });
-
-  // materialize the Domain's resolvedAddress field iff exists and is set to this Resolver
-  if (domain && domain.resolverId === id) {
-    await context.db.update(schema.domain, { id: node }).set({ resolvedAddressId: null });
-  }
-
-  await upsertResolver(context, {
-    id,
-    domainId: node,
-    address: event.log.address,
-
-    // clear out the resolver's info
-    addrId: null,
-    contentHash: null,
-    coinTypes: null,
-    texts: null,
-  });
-
-  // log ResolverEvent
-  await context.db.insert(schema.versionChanged).values({
-    ...sharedEventValues(event),
-    resolverId: id,
-    version: newVersion,
-  });
-}
-
-export async function handleDNSRecordChanged({
-  context,
-  event,
-}: {
-  context: Context;
-  event: EventWithArgs<{
-    node: Hex;
-    name: Hex;
-    resource: number;
-    record: Hex;
-  }>;
-}) {
-  // subgraph ignores
-}
-
-export async function handleDNSRecordDeleted({
-  context,
-  event,
-}: {
-  context: Context;
-  event: EventWithArgs<{
-    node: Hex;
-    name: Hex;
-    resource: number;
-    record?: Hex;
-  }>;
-}) {
-  // subgraph ignores
-}
-
-export async function handleDNSZonehashChanged({
-  context,
-  event,
-}: {
-  context: Context;
-  event: EventWithArgs<{ node: Hex; zonehash: Hex }>;
-}) {
-  // subgraph ignores
-}
+export const makeResolverHandlers = (ownedName: OwnedName) => {
```

Comment:
I had to wrap multiple exports into a factory function that would allow passing extra context to those previously-exported functions. Haven't changed logic of any of these functions.